helix-core/src/main/java/org/apache/helix/controller/rebalancer/waged/model/AssignableNode.java

```diff
@@ -88,11 +88,15 @@ private void refresh(ClusterConfig clusterConfig, InstanceConfig instanceConfig,
       Collection<AssignableReplica> existingAssignment) {
     reset();
 
-    _currentCapacity.putAll(instanceConfig.getInstanceCapacityMap());
+    Map<String, Integer> instanceCapaity = instanceConfig.getInstanceCapacityMap();
```

Comment:
Do:
```
_maxCapacity = instanceConfig.getInstanceCapacityMap();
if (_maxCapacity != null && _maxCapacity.isEmpty()) {
  _maxCapacity = clusterConfig.getDefaultInstanceCapacityMap();
}
..
_currentCapacity.putAll(_maxCapacity);
```
Also, do we want to recompute currentCapacity from scratch every rebalance? We certainly could for an MVP but just making sure (this optimization could come later).

Reply:
getInstanceCapacityMap won't return null. This is guaranteed and the unit test covers it.

You are right. That one of the reasons why I tried to decouple the model classes as far as I can. In this case, if one instance is not changed, we can just keep it. That can be done in the Cluster Model Provder.
It is in my plan, but will be future.

Reply:
Discussed with Hunter offline. I got the point now. Basically, we can potentially save the temp variable instanceCapaity with the suggested code. This is a valid point. But for this refresh logic, I prefer not to set and read the private field in the same function. Just a way to avoid confusing and reduce potential bug.